osquery/tables/system/linux/processes.cpp

```diff
@@ -55,12 +60,46 @@ inline std::string readProcLink(const std::string& attr,
   char link_path[PATH_MAX] = {0};
   auto bytes = readlink(attr_path.c_str(), link_path, sizeof(link_path) - 1);
   if (bytes >= 0) {
+    link_path[bytes] = 0; // readlink does not append NUL
```

Comment:
Isn't this `link_path` initialized with `{0}`, is there a need to write a final `0`? Does the `::readlink` call potentially add garbage?